CMakeLists.txt

```diff
@@ -120,15 +120,22 @@ sitk_legacy_naming(SimpleITK_INT64_PIXELIDS)
 
 
 
-set(SimpleITK_4D_IMAGES_DEFAULT ON)
+set(SimpleITK_MAX_DIMENSION_DEFAULT 4)
 # 1900 = VS 14.0 (Visual Studio 2015)
 if(MSVC AND MSVC_VERSION VERSION_LESS 1900)
-  set( SimpleITK_4D_IMAGES_DEFAULT OFF )
-  message("disable")
+  set( SimpleITK_MAX_DIMENSION_DEFAULT 3 )
 endif()
-option( SimpleITK_4D_IMAGES "Add Image and I/O support for four spatial dimensions." ${SimpleITK_4D_IMAGES_DEFAULT} )
-mark_as_advanced( SimpleITK_4D_IMAGES )
-sitk_legacy_naming(SimpleITK_4D_IMAGES)
+if (DEFINED SimpleITK_4D_Image)
+  message(WARNING "SimpleITK variable \"SimpleITK_4D_Image\" is \
```

Comment:
Maybe say "set the \"SimpleITK_MAX_DIMENSION\" to 4 instead."?

Reply:
We only know that SimpleITK_4D_Image is defined and not is value, so we don't know what value to recommend it be set to.